src/piped/actions.clj

```diff
@@ -16,10 +16,10 @@
 
 (defn spawn-nacker
   "Nacks batches of messages."
-  [client input-chan]
+  [client input-chan backoff-fn]
```

Comment:
This breaks backwards compatibility of a public function. You could change `spawn-nacker` into a multi-arity function that accepts an optional third argument `opts` map that contains a `nack-delay-fn` that will be used if defined. default to an fn that returns 0 if nothing is defined.

Reply:
I prefer nack-delay-fn because it feels more accurate. It may contain backoff code and maybe not. Maybe certain queues would have their own nack-delay. I do know it will return a number that will delay the "nacking" of the message regardless of the logic behind that number.

Reply:
I think regardless of what the function contains it leads to a backoff given that it influences the `VisibilityTimeout` of the message. Maybe the wording is confusing and it could be better as something like `nack-visibility-timeout-fn`? 

I'm not sure how far out the abstraction/hiding of details from the client is useful though. As a user `backoff-fn` is far more intuitive in what it leads to while something like `nack-delay-fn` or `nack-visibility-timeout-fn` are maybe a bit more 'in the weeds' for a user. 

Again I'll defer to @RutledgePaulV since this is their library and I have no real qualms/claim to the name